### PR TITLE
Sort grouped things by the full report name

### PR DESCRIPTION
**Pull request by tgolen**
Reported by @coleaeason here: https://expensify.slack.com/archives/C01GTK53T8Q/p1663860286166959

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/229779

### Tests
1. Make sure you are in default view mode
2. Make sure you have several pinned chats
3. Verify the group of pinned chats are sorted alphabetically in the LHN
4. Pin a new chat
5. Verify it gets added in the proper alphabetical order in the LHN
2. Make sure you have draft chats
3. Verify the group of draft chats are sorted alphabetically in the LHN
4. Add a new draft to a new report
5. Verify it gets added in the proper alphabetical order in the LHN

- [ ] Verify that no errors appear in the JS console

#### Web
image

### CODE REVIEW

**Review by neil-marcellini — COMMENTED** [review]
Looking good. Would you please also add automated and manual test for drafts sorted alphabetically?

**Review by alex-mechler — COMMENTED** [review]
That, and travis is currently failing

**Comment by tgolen** [comment]
Updated with tests for drafts also.

@alex-mechler the failing tests are OK since this is not a contributor issue.

**Review by alex-mechler — DISMISSED** [review]
LGTM, I tried doing a little digging to see where we decided that we didn't have to fill out the check list as internal engineers, but couldn't find anything. Do you have a link to that handy?

File: tests/unit/LHNOrderTest.js [hunk 1]
```diff
@@ -499,6 +499,104 @@ describe('Sidebar', () => {
...
+                });
+        });
+
+        it('alphabetizes all the chats that have drafts', () => {
+            const sidebarLinks = getDefaultRenderedSidebarLinks();
+
+            return waitForPromisesToResolve()
+
+                // Given the sidebar is rendered in default mode
+                // with all reports being pinned
```
Comment (neil-marcellini) [inline tests/unit/LHNOrderTest.js:558]: NAB: `// with all reports having drafts`

**Review by neil-marcellini — DISMISSED** [review]
Thanks for adding that test! Looks good. One NAB on the comment.

**Comment by tgolen** [comment]
> Do you have a link to that handy?

https://github.com/Expensify/Expensify/issues/230010

**Review by neil-marcellini — APPROVED** [review]

**Comment by neil-marcellini** [comment]
We can ignore these checklists since it is not a contributor issue https://github.com/Expensify/App/pull/11211#issuecomment-1255441878. We have an issue open to fix that.

**Pull request merged**